### DISCUSSION TITLE
Making demographic data easier to work with

### DISCUSSION THREAD

**Discussion by ghop02** [covid-projections/can-scrapers#209] [Ideas]
Wanted to see what ways would make dealing with demographic data a little bit easier.

https://github.com/covid-projections/can-scrapers/compare/demographic-data-demo?expand=1

I would like to flesh this example out a bit more but just wondering if there's code we're already using to mold demographic data to our liking and if we could solidify a couple patterns

**Comment by sglyon** [comment]
This is a great initiative. 

Right now demographic data is clunky (at best!) to normalize.  Given current tools/infrastructure, we resort to long dicts mapping column names to CMU: https://github.com/covid-projections/can-scrapers/blob/e3a933ec7fb22ccdbb7c0ddd1a9545c967dfa213/can_tools/scrapers/official/DC/dc_cases.py

I see the new `included_demographic_columns` argument to the `extract_CMU`  method. It seems like that is duplicating some of the behavior/logic of the `columns` argument to that method.

**Reply by ghop02** [reply]
I think the `columns` is more all of the columns in the CMU object.  in this case, the separate `included_demographics_columns` is needed because data frames may not have all of the demographic columns and if not they take it from the value in the CMU. However, it's possible a world could exist where that function just recognized if there are demographic columns and not need to pass it explicitly
